language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java

```diff
@@ -44,16 +53,74 @@ public TransactionFilterOuterClass.InterfaceFilter toProto(Identifier interfaceI
     }
 
     static Interface fromProto(TransactionFilterOuterClass.InterfaceFilter proto) {
-      return includeInterfaceView(proto.getIncludeInterfaceView());
+      return includeInterfaceView(
+          proto.getIncludeInterfaceView(), proto.getIncludeCreateEventPayload());
     }
 
     Interface merge(Interface other) {
-      return includeInterfaceView(includeInterfaceView || other.includeInterfaceView);
+      return includeInterfaceView(
+          includeInterfaceView || other.includeInterfaceView,
+          includeCreateEventPayload || other.includeCreateEventPayload);
     }
 
     @Override
     public String toString() {
-      return "Filter.Interface{" + "includeInterfaceView=" + includeInterfaceView + '}';
+      return "Filter.Interface{"
+          + "includeInterfaceView="
+          + includeInterfaceView
+          + ", includeCreateEventPayload="
+          + includeCreateEventPayload
+          + '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      Filter.Interface that = (Filter.Interface) o;
+      return Objects.equals(includeInterfaceView, that.includeInterfaceView)
+          && Objects.equals(includeCreateEventPayload, that.includeCreateEventPayload);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(includeInterfaceView, includeCreateEventPayload);
+    }
+  }
+
+  public static final class Template {
```

Comment:
Also probably better as an enum.

```suggestion
  public static enum Template {
    INCLUDE_PAYLOAD(true),
    HIDE_PAYLOAD(false);
```

Reply:
Done